colorbleed/__init__.py

```diff
@@ -23,3 +25,36 @@ def uninstall():
     pyblish.deregister_plugin_path(PUBLISH_PATH)
     avalon.deregister_plugin_path(avalon.Loader, LOAD_PATH)
 
+
+def _get_host_name():
+
+    _host = avalon.registered_host()
+    # This covers nested module name like avalon.maya
+    return _host.__name__.rsplit(".", 1)[-1]
+
+
+def collect_container_metadata(container):
+    """Add additional data based on the current host
+
+    If the host application's lib module does not have a function to inject
+    additional data it will return the input container
+
+    Args:
+        container (dict): collection if representation data in host
+
+    Returns:
+        generator
+    """
+
+    host_name = _get_host_name()
+
+    # This will cover nested module names like avalon.maya
+    package_name = "{}.{}.lib".format(__name__, host_name)
+    hostlib = importlib.import_module(package_name)
+
+    if not hasattr(hostlib, "get_additional_data"):
+        print("{} has no function called "
+              "get_additional_data".format(package_name))
+        return container
```

Comment:
This shouldn't return the container - correct? It should then return an empty dictionary?

Reply:
You are correct! If we return a container we would update the new item with the container. Even though that would not be a problem currently it will reduce computing time ;)